File: app/java/src/main/java/com/example/production/domain/model/plan/Allocation.java
package com.example.production.domain.model.plan;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Allocation {
    private Integer id;
    private Integer requirementId;
    private AllocationType allocationType;
    private Integer orderId;
    private LocalDate allocationDate;
    private BigDecimal allocatedQuantity;
    private String locationCode;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Plan/Allocation.cs
namespace ProductionManagement.Domain.Models.Plan;

/// <summary>
/// 引当情報
/// </summary>
public class Allocation
{
    private string _allocationTypeValue = "在庫";

    public int Id { get; set; }

    public int RequirementId { get; set; }

    /// <summary>
    /// 引当区分（データベース格納用の日本語文字列）
    /// </summary>
    public string AllocationTypeValue
    {
        get => _allocationTypeValue;
        set => _allocationTypeValue = value;
    }

    /// <summary>
    /// 引当区分（ドメインロジック用の Enum）
    /// </summary>
    public AllocationType AllocationType
    {
        get => AllocationTypeExtensions.FromDisplayName(_allocationTypeValue);
        set => _allocationTypeValue = value.GetDisplayName();
    }

    public int? OrderId { get; set; }

    public DateOnly AllocationDate { get; set; }

    public decimal AllocatedQuantity { get; set; }

    public required string LocationCode { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    // リレーション
    public Requirement? Requirement { get; set; }
    public Order? Order { get; set; }
}
